src/legacy/ui/public/field_editor/field_editor.tsx

```diff
@@ -135,48 +180,54 @@ export class FieldEditor extends PureComponent {
   }
 
   async init() {
-    const { $http } = this.props.helpers;
+    const { getHttpStart } = this.props.helpers;
     const { field } = this.state;
     const { indexPattern } = this.props;
 
-    const getEnabledScriptingLanguages = new GetEnabledScriptingLanguagesProvider($http);
-    const enabledLangs = await getEnabledScriptingLanguages();
+    const enabledLangs = await getEnabledScriptingLanguages(await getHttpStart());
     const scriptingLangs = intersection(
       enabledLangs,
       union(this.supportedLangs, this.deprecatedLangs)
     );
-    field.lang = scriptingLangs.includes(field.lang) ? field.lang : undefined;
+    field.lang = field.lang && scriptingLangs.includes(field.lang) ? field.lang : undefined;
 
-    const fieldTypes = get(FIELD_TYPES_BY_LANG, field.lang, DEFAULT_FIELD_TYPES);
+    const fieldTypes = get(FIELD_TYPES_BY_LANG, field.lang || '', DEFAULT_FIELD_TYPES);
     field.type = fieldTypes.includes(field.type) ? field.type : fieldTypes[0];
 
     const fieldFormats = getFieldFormats();
-    const DefaultFieldFormat = fieldFormats.getDefaultType(field.type, field.esTypes);
+    const DefaultFieldFormat = fieldFormats.getDefaultType(
+      field.type as KBN_FIELD_TYPES,
+      field.esTypes as ES_FIELD_TYPES[]
+    );
 
     this.setState({
       isReady: true,
-      isCreating: !indexPattern.fields.getByName(field.name),
-      isDeprecatedLang: this.deprecatedLangs.includes(field.lang),
+      isCreating: !indexPattern.fields.find(f => f.name === field.name),
+      isDeprecatedLang: this.deprecatedLangs.includes(field.lang || ''),
       errors: [],
       scriptingLangs,
       fieldTypes,
-      fieldTypeFormats: getFieldTypeFormatsList(field, DefaultFieldFormat),
+      fieldTypeFormats: getFieldTypeFormatsList(
+        field,
+        DefaultFieldFormat as FieldFormatInstanceType
+      ),
       fieldFormatId: get(indexPattern, ['fieldFormatMap', field.name, 'type', 'id']),
       fieldFormatParams: field.format.params(),
     });
   }
 
-  onFieldChange = (fieldName, value) => {
+  onFieldChange = (fieldName: string, value: string | number) => {
     const { field } = this.state;
+    // @ts-ignore
     field[fieldName] = value;
```

Comment:
Could we avoid `@ts-ignore` here, maybe below would work

```suggestion
    (field as any)[fieldName] = value;
```